sim/battle.ts

```diff
@@ -573,8 +564,7 @@ export class Battle extends Dex.ModdedDex {
 			hasRelayVar = false;
 		}
 
-		// @ts-ignore
-		if (effect.effectType === 'Status' && target.status !== effect.id) {
+		if (effect.effectType === 'Status' && (target instanceof Pokemon) && target.status !== effect.id) {
```

Comment:
```suggestion
		if (effect.effectType === 'Status' && (target as Pokemon).status !== effect.id) {
```

We try to use duck typing rather than `instanceof`.

Reply:
`singleEvent` does this (I've already tried to shard the code between the suppression checks in `runEvent` and `singleEvent` but ran into problems)

Reply:
I'm tentatively fine with either resolution.

Reply:
Going to stick with the `instanceof` like with `singleEvent`.